### PR TITLE
Arc 1787 clean up tslint warnings

### PR DESCRIPTION
**Pull request by gxueatlassian**
**What's in this PR?**
1. Turn on errors for some of the rules that's previously only warnings
2. Fix some low hangings fruits for warnings.

**Why**
Remove the noises as much as possible.

**Added feature flags**
N/A

**Affected issues**  
ARC-1787

**How has this been tested?**  
Unit test

**Whats Next?**
Further remove typings warnings.

prev

new

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -59,8 +59,25 @@
...
 		"require-await": "off",
-		"@typescript-eslint/no-floating-promises": ["error"]
-	}
+		"@typescript-eslint/no-floating-promises": ["error"],
+		"jest/expect-expect": [
+			"error",
+			{
+				"assertFunctionNames": [
+					"expect",
+					"supertest.**.expect",
```
Comment (gxueatlassian) [inline .eslintrc.json:70]: this is for making `supertest(app).get('/xxx').status(200)` passing linting.

File: .eslintrc.json [hunk 2]
```diff
@@ -59,8 +59,25 @@
...
+					"*.expect"
```
Comment (gxueatlassian) [inline .eslintrc.json:71]: This is for making `someInternaMethod(xxx).expect(200)` to pass. We'll trust the author as long as they wrote `expect` somewhere.

File: .eslintrc.json [hunk 3]
```diff
@@ -59,8 +59,25 @@
...
+				]
+			}
+		]
+	},
+	"overrides": [{
+		"files": [ "src/**/*.test.ts" ],
+		"rules": {
+			"@typescript-eslint/no-explicit-any": "off",
+			"@typescript-eslint/no-non-null-assertion": "off"
```
Comment (gxueatlassian) [inline .eslintrc.json:80]: This is for making `DatabaseStateCreator.gitHubServerApp!.gitHubAppId` pass eslint.
We do want the `!` after the `gitHubServerApp` for fail-fast. So I think it is better to allow it in tests files.

In runtime, allowing `!` may cause a bug, while in test time, allow `!` should fail-fast (which is better).

File: src/sync/branch.test.ts [hunk 4]
```diff
@@ -119,7 +119,7 @@ describe("sync/branches", () => {
 				});
 		};
 
-		const nockBranchRequest = (response, variables?: Record<string, any>) =>
+		const nockBranchRequest = (response: object, variables?: Record<string, unknown>) =>
```
Comment (gxueatlassian) [inline src/sync/branch.test.ts:122]: `unknown` is better than `any` as the ts prevents code from using any fields from that object.

**Review by mboudreau — APPROVED** [review]
Looks good :+1:

**Pull request merged**